content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md

```diff
@@ -0,0 +1,175 @@
+---
+layout: blog
+title: "Try the new gRPC probes in 1.24: now in beta."
+date: 2022-04-15
+slug: grpc-probes-now-in-beta
+---
+
+With Kubernetes 1.24 the gRPC probes functionality entered beta and is available by default.
+Now you can configure startup, liveness, and readiness probes for your gRPC app
+without exposing http endpoint or using an extra executable. All built-in to the Kubernetes.
+
+## Some history
+
+Kubernetes allows to control the container lifecycle from inside the app
+by checking http response, running executable, or simply connecting to a tcp port.
+
+Those checks are often enough for the most of apps. For gRPC apps, it is easy
+to repurpose the `exec` probe to use it for gRPC health checking. This solution
+was described in the blog post "[Health checking gRPC servers on Kubernetes](https://kubernetes.io/blog/2018/10/01/health-checking-grpc-servers-on-kubernetes/)"
+and used for a long time. Commonly used tool to enable this was created on [Aug 21, 2018](https://github.com/grpc-ecosystem/grpc-health-probe/commit/2df4478982e95c9a57d5fe3f555667f4365c025d) with
+the first release at [Sep 19, 2018](https://github.com/grpc-ecosystem/grpc-health-probe/releases/tag/v0.1.0-alpha.1).
+
+This approach for gRPC apps health checking is very popular. There are [3,626 Dockerfiles](https://github.com/search?l=Dockerfile&q=grpc_health_probe&type=code)
+with the `grpc_health_probe` and [6,621 yaml](https://github.com/search?l=YAML&q=grpc_health_probe&type=Code) files that are discovered with the
+basic search on GitHub (at the moment of writing). This is good indication of the tool popularity
+and the need to support this natively.
+
+With the Kubernetes 1.23, native support for gRPC health probing was introduced as alpha feature.
+The feature is not at beta and enabled by default.
```

Comment:
```suggestion
Kubernetes v1.23 introduced an alpha-quality implementation of native support for
querying a workload status using gRPC.
Because it was an alpha feature, this was disabled by default for the v1.23 release.
```